packages/salesforcedx-vscode-core/src/commands/commands.ts

```diff
@@ -48,6 +48,31 @@ export interface ParametersGatherer<T> {
   gather(): Promise<CancelResponse | ContinueResponse<T>>;
 }
 
+export class CompositeParametersGatherer<T> implements ParametersGatherer<T> {
+  protected readonly gatherers: ParametersGatherer<any>[];
+  public constructor(...gatherers: ParametersGatherer<any>[]) {
+    this.gatherers = gatherers;
+  }
+  public async gather(): Promise<CancelResponse | ContinueResponse<T>> {
+    const aggregatedData: any = {};
+    for (const gatherer of this.gatherers) {
+      const input = await gatherer.gather();
+      if (input.type === 'CONTINUE') {
+        Object.keys(input.data).map(
+          key => (aggregatedData[key] = input.data[key])
+        );
+      } else {
+        return {
+          type: 'CANCEL'
+        };
+      }
+    }
+    return {
+      type: 'CONTINUE',
+      data: aggregatedData
+    };
+  }
+}
 export class EmptyParametersGatherer implements ParametersGatherer<{}> {
```

Comment:
The prettier formatter won't add automatically add new lines.
So you should add a new line here to separate the two classes.